src/toolchain.rs

```diff
@@ -1,29 +1,58 @@
 use anyhow::{bail, Result};
 use std::fmt;
+use std::fs::{remove_dir_all, remove_file};
 use std::path::PathBuf;
 use std::str::FromStr;
 use tracing::info;
 
+use crate::component;
 use crate::download::{download_file_and_unpack, link_to_fuelup, unpack_bins, DownloadCfg};
 use crate::ops::fuelup_self::self_update;
-use crate::path::{ensure_dir_exists, fuelup_bin, fuelup_bin_dir, toolchain_bin_dir};
+use crate::path::{
+    ensure_dir_exists, fuelup_bin, fuelup_bin_dir, settings_file, toolchain_bin_dir,
+};
+use crate::settings::SettingsFile;
+
+pub const RESERVED_TOOLCHAIN_NAMES: &[&str] = &["latest", "nightly"];
```

Comment:
Perhaps we should conservatively add "stable" and "beta" so that user's toolchains don't break if we eventually introduce them?

Also, it could be useful to have each of these names as `const`s for use independently elsewhere.